spinalcordtoolbox/scripts/sct_run_batch.py

```diff
@@ -446,26 +474,15 @@ def reset_streams():
                 for isess in session_dirs:
                     subject_dirs.append(os.path.join(isub, isess))
 
-    # Handle inclusion lists
-    assert not ((arguments.include is not None) and (arguments.include_list is not None)),\
-        'Only one of `include` and `include-list` can be used'
-
-    if arguments.include is not None:
-        subject_dirs = [f for f in subject_dirs if re.search(arguments.include, f) is not None]
-
-    if arguments.include_list is not None:
-        # TODO decide if we should warn users if one of their inclusions isn't around
-        subject_dirs = [f for f in subject_dirs if f in arguments.include_list]
-
-    # Handle exclusions
-    assert not ((arguments.exclude is not None) and (arguments.exclude_list is not None)),\
-        'Only one of `exclude` and `exclude-list` can be used'
+    if (arguments.include is not None) and (arguments.include_list is not None):
+        parser.error('Only one of `include` and `include-list` can be used')
```

Comment:
Thanks for turning this into a proper error rather than an `assert`. I've always liked the description in [this style guide](https://google.github.io/styleguide/pyguide.html#244-decision), that asserts should only be used for internal invariants (because they [may be optimized out](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement)).

Reply:
> I've always liked the description in [this style guide](https://google.github.io/styleguide/pyguide.html#244-decision),

Aside: Thank you for linking this style guide! It's interesting to see the internal perspectives of a mega-corporation like Google. There's a whole lot to unpack, but it seems like a valuable resource. :)

> (because they [may be optimized out](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement)).

Huh. I'd never heard of [`-O`](https://docs.python.org/3/using/cmdline.html#cmdoption-O) before.

> Remove assert statements and any code conditional on the value of [`__debug__`](https://docs.python.org/3/library/constants.html#debug__). Augment the filename for compiled ([bytecode](https://docs.python.org/3/glossary.html#term-bytecode)) files by adding .opt-1 before the .pyc extension (see [PEP 488](https://www.python.org/dev/peps/pep-0488)). See also [PYTHONOPTIMIZE](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONOPTIMIZE).

I wonder how this would be used in practice... :thinking:

Reply:
I think it's pretty rare to use `-O` for Python code, so `assert` statements are very likely to run in practice. This is definitely a tiny, technical, theoretical detail, essentially trivia.